libflux/src/flux/semantic/builtins.rs

```diff
@@ -0,0 +1,338 @@
+                 "second" => Node::Builtin("forall [] (t: time) -> int"),
+                 "minute" => Node::Builtin("forall [] (t: time) -> int"),
+                 "hour" => Node::Builtin("forall [] (t: time) -> int"),
+                 "weekDay" => Node::Builtin("forall [] (t: time) -> int"),
+                 "monthDay" => Node::Builtin("forall [] (t: time) -> int"),
+                 "yearDay" => Node::Builtin("forall [] (t: time) -> int"),
+                 "month" => Node::Builtin("forall [] (t: time) -> int"),
+                 "year" => Node::Builtin("forall [] (t: time) -> int"),
+                 "week" => Node::Builtin("forall [] (t: time) -> int"),
+                 "quarter" => Node::Builtin("forall [] (t: time) -> int"),
+                 "millisecond" => Node::Builtin("forall [] (t: time) -> int"),
+                 "microsecond" => Node::Builtin("forall [] (t: time) -> int"),
+                 "nanosecond" => Node::Builtin("forall [] (t: time) -> int"),
+                 "truncate" => Node::Builtin("forall [] (t: time, unit: duration) -> time"),
+            }),
+            "experimental" => Node::Package(maplit::hashmap! {
+                 "bigtable" => Node::Package(maplit::hashmap! {
+                     "from" => Node::Builtin("forall [t0] (token: string, project: string, instance: string, table: string) -> [t0]"),
+                 }),
+                 "http" => Node::Package(maplit::hashmap! {
+                     "get" => Node::Builtin("forall [t0, t1] (url: string, ?headers: t0, ?timeout: duration) -> {statusCode: int | body: bytes | headers: t1}"),
+                 }),
+                 "mqtt" => Node::Package(maplit::hashmap! {
+                     "to" => Node::Builtin(r#"
+                         forall [t0, t1] (
+                             <-tables: [t0],
+                             broker: string,
+                             ?topic: string,
+                             ?message: string,
+                             ?qos: int,
+                             ?clientid: string,
+                             ?username: string,
+                             ?password: string,
+                             ?name: string,
+                             ?timeout: duration,
+                             ?timeColumn: string,
+                             ?tagColumns: [string],
+                             ?valueColumns: [string]
+                         ) -> [t1]
+                     "#),
+                 }),
+                 "prometheus" => Node::Package(maplit::hashmap! {
+                     "scrape" => Node::Builtin("forall [t0] (url: string) -> [t0]"),
+                 }),
+                 "addDuration" => Node::Builtin("forall [] (d: duration, to: time) -> time"),
+                 "subDuration" => Node::Builtin("forall [] (d: duration, from: time) -> time"),
+                 "group" => Node::Builtin("forall [t0, t1] (<-tables: [t0], mode: string, columns: [string]) -> t1"),
```

Comment:
Group doesn't modify its input records right? Should this be `forall [t0] (<-tables: [t0], ... ) -> t0`?

Reply:
Yes, you are right.